src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -33,8 +33,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <FullRuntimeName Condition="'$(ServerGarbageCollection)' == 'true'">libRuntime.ServerGC</FullRuntimeName>
 
       <CrossCompileRid />
-      <CrossCompileRid Condition="!$(RuntimeIdentifier.EndsWith('-$(_hostArchitecture)'))">$(RuntimeIdentifier)</CrossCompileRid>
-      <CrossCompileRid Condition="'$(CrossCompileRid)' == '' and '$(_linuxLibcFlavor)' == 'bionic'">$(RuntimeIdentifier)</CrossCompileRid>
+      <CrossCompileRid Condition="!$(RuntimeIdentifier.StartsWith('$(_hostOS)-')) or !$(RuntimeIdentifier.EndsWith('-$(_hostArchitecture)'))">$(RuntimeIdentifier)</CrossCompileRid>
```

Comment:
Would this break targeting linux-bionic-x64 from linux-x64? (_hostOs is going to be linux and linux-bionic-x64 starts with that, and ends with x64)

I don't have a good sense for how the fix should look like. Intuitively this should be _hostOs != _targetOs, but we normalize _targetOs to linux, so it's more complicated than that.

Reply:
Fixed, also handled a corner case for source-build overridden RuntimeIdentifier.